core/modules/layout/layout.admin.inc

```diff
@@ -1554,6 +1570,114 @@ function layout_block_remove_page(Layout $layout, Block $block, $renderer_name)
   }
 }
 
+/**
+ * Submit handler for layout_settings_form() that lets the user add a condition.
+ */
+function layout_settings_form_context_add($form, &$form_state) {
+  // Remove destination flags when adding conditions, otherwise the user will
+  // be taken to the destination instead of adding a condition. This only
+  // affects non-JS behavior.
+  if (isset($_GET['destination'])) {
+    unset($_GET['destination']);
+  }
+  $form_state['redirect'] = 'admin/structure/layouts/manage/' . $form_state['layout']->name . '/context/add';
+}
+
+/**
+ * Form callback; Displays form for adding new conditions to a layout or block.
+ *
+ * @param Layout $layout
+ *   The layout to which a condition is being added.
+ * @param Block|null $block
+ *   If applying a condition to a particular block, the block instance.
+ * @param LayoutMenuItem|null $menu_item
+ *   If applying a condition to an menu item, the menu item object.
+ * @param int|null $condition_id
+ *   The integer ID of the condition being configured. If adding a new
+ *   condition, no value will be passed in. Note that this value may be a
+ *   FALSE-like value, such as 0, indicating the first condition is being
+ *   configured, which is different than a NULL value indicating a new
+ *   condition is being added.
+ *
+ * @ingroup forms
+ */
+function layout_context_add_form($form, &$form_state, Layout $layout = NULL, Block $block = NULL, LayoutMenuItem $menu_item = NULL, $context_id = NULL) {
+  form_load_include($form_state, 'inc', 'layout', 'layout.admin');
+  $form_state['layout'] = $layout;
+  $form_state['block'] = $block;
+  $form_state['menu_item'] = $menu_item;
+  $form_state['context_id'] = $context_id;
+
+  $form['help'] = array(
+    '#type' => 'help',
+    '#markup' => 'help',
+  );
+
+  $handler = isset($layout->contexts[$context_id]) ? $layout->contexts[$context_id] : NULL;
+
+  $all_context_info = _layout_get_all_info('layout_context');
+  backdrop_sort($all_context_info, array('title'));
+  $form['context'] = array(
+    '#type' => 'select',
+    '#title' => t('Contexts'),
+    '#options' => array(),
+    '#required' => TRUE,
+    '#default_value' => isset($handler->plugin) ? $handler->plugin : NULL,
+    '#parents' => array('context'),
+    '#ajax' => array(
+      'wrapper' => 'context_settings',
+      'callback' => 'layout_context_ajax_style',
+    ),
+  );
+  foreach ($all_context_info as $context_info) {
+    $form['context']['#options'][$context_info['name']] = $context_info['title'];
+  }
```

Comment:
As a non experienced developer, I'd expect this loop to come before the select element, setting the values of say a `$context_options` array variable, and then use that array instead of the empty array in `'#options' => array(),` earlier.